lib/postcssPlugin.js

```diff
@@ -1,26 +1,41 @@
-// @ts-nocheck
 'use strict';
 
 const createStylelint = require('./createStylelint');
 const path = require('path');
-const postcss = require('postcss');
 
-module.exports = postcss.plugin('stylelint', (options = {}) => {
-	const tailoredOptions = options.rules ? { config: options } : options;
-	const stylelint = createStylelint(tailoredOptions);
+/** @typedef {import('stylelint').PostcssPluginOptions} PostcssPluginOptions */
+/** @typedef {import('stylelint').StylelintConfig} StylelintConfig */
 
-	return (root, result) => {
-		let filePath = options.from;
```

Comment:
It's not clear what `options.from` supposed to be used for. It's not documented, and there are no tests that would use it. Couldn't find anything in Git history. [Trace](https://github.com/stylelint/stylelint/blame/ebc9e6a69bed159982eee5e0678d29b5b23747c6/lib/postcssPlugin.js) is lost on `Drop babel` commit.